File: backend/services/content-service/internal/domain/document.go
package domain

import "time"

// Document is a PDF attached to a territory. It has no scene position and no
// slug — it is identified by ID and its bytes are served from BlobStore via
// the asset service at /api/assets/{SourceBlobHash}.
type Document struct {
	ID             int64
	TerritorySlug  string
	Title          string
	SourceBlobHash string
	CreatedAt      time.Time
}

File: frontend-v2/src/entities/document/model/document.ts
/** A PDF attached to a territory. Served as-is from BlobStore, not converted, not anchored in the scene. */
export type Document = {
  id: number;
  territorySlug: string;
  title: string;
  sourceBlobHash: string;
  createdAt: string;
};

export type DocumentCreate = { title: string; sourceBlobHash: string };

/** The mock's row and window print the file name; the gateway keeps a title. */
export const documentFileName = (d: Document) => d.title;
